File: papi/lista1/exercicio8.py
while True:
    try:
        valor1 = float(input("Digite o valor 1: "))
        break
    except ValueError:
        print("Por favor digite um número válido")

while True:
    try:
        valor2 = float(input("Digite o valor 2: "))
        break
    except ValueError:
        print("Por favor digite um número válido")

while True:
    try:
        valor3 = float(input("Digite o valor 3: "))
        break
    except ValueError:
        print("Por favor digite um número válido")

valores_ordenados = [valor1, valor2, valor3]
valores_ordenados.sort()
print("Os valores ordenados são:", valores_ordenados[0], ",", valores_ordenados[1], ",", valores_ordenados[2])

File: papi/lista1/exercicio8.kt
package lista1

fun main() {
    println("Vamos ordenar 3 valores")
    var valores: ArrayList<Double> = ArrayList();

    while (true) {
        try {
            print("Digite o primeiro valor: ")
            val input = readLine()
            valores.add(input!!.toDouble())
            break
        } catch (e: NumberFormatException) {
            println("Por favor digite um valor numérico, com o ponto como separador decimal")
        }
    }

    while (true) {
        try {
            print("Digite o segundo valor: ")
            val input = readLine()
            valores.add(input!!.toDouble())
            break
        } catch (e: NumberFormatException) {
            println("Por favor digite um valor numérico, com o ponto como separador decimal")
        }
    }

    while (true) {
        try {
            print("Digite o terceiro valor: ")
            val input = readLine()
            valores.add(input!!.toDouble())
            break
        } catch (e: NumberFormatException) {
            println("Por favor digite um valor numérico, com o ponto como separador decimal")
        }
    }

    valores.sort()
    print("Um ArrayList com os valores ordenados é igual a: ")
    println(valores)

}